src/panels/lovelace/editor/config-elements/hui-action-base-card-editor.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  customElement,
+  internalProperty,
+  LitElement,
+  property,
+  query,
+} from "lit-element";
+
+import { fireEvent, HASSDomEvent } from "../../../../common/dom/fire_event";
+import { ActionConfig } from "../../../../data/lovelace";
+import { HomeAssistant } from "../../../../types";
+import { LovelaceCardEditor } from "../../types";
+import { GUIModeChangedEvent } from "../types";
+import type { HuiElementEditor } from "../hui-element-editor";
+import type { EditActionEvent } from "../../components/hui-actions-editor";
+
+import "../hui-element-editor";
+
+@customElement("hui-action-base-card-editor")
+export class HuiActionBaseCardEditor extends LitElement
```

Comment:
I see why you did this, but not really happy with it, we have to think about a better way.

Reply:
Can you elaborate on which aspect you don't like about it?

Reply:
Maybe we should make it a mixin, but the name of the properties/functions now is too general, and the logic too much separated between this element and the card editor.

Reply:
What benefit does a mixin give us over extending a class?

Reply:
That we could add it to more than just 1 class, if we ever decide we want an action editor and a theme editor, we can just add mixins to the editors that need themes and/or actions